exercises/largest-series-product/largest_series_product_test.go

```diff
@@ -2,27 +2,39 @@ package lsproduct
 
 import "testing"
 
-const targetTestVersion = 3
+const targetTestVersion = 4
 
-func TestLargestSeriesProduct(t *testing.T) {
+func TestTestVersion(t *testing.T) {
 	if testVersion != targetTestVersion {
 		t.Fatalf("Found testVersion = %v, want %v", testVersion, targetTestVersion)
 	}
+}
+
+func TestLargestSeriesProduct(t *testing.T) {
 	for _, test := range tests {
 		p, err := LargestSeriesProduct(test.digits, test.span)
-		switch {
-		case err != nil:
-			if test.ok {
+		if !test.ok { // expect error
+			// check if err is of error type
+			var _ error = err
```

Comment:
> If the test case is for good path(non-negative) then we are interested in 'p'(product) only.

Yup.

> err should be checked only when error is expected

It's true that whether `err` is nil or not should only be checked when an error is expected.

We are free to check whether `err` is of type `error` anywhere though. The argument here is that maybe we should have the `var _ error = err` line earlier so that it is clearer to the student that the `err` needs to be of type `error`, instead of farther down in the file. what do you think about this?